pubsub/subscriptions/subscription_test.go

```diff
@@ -779,6 +780,36 @@ func TestCreateBigQuerySubscription(t *testing.T) {
 	}
 }
 
+func TestCreateCloudStorageSubscription(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	tc := testutil.SystemTest(t)
+	client := setup(t)
+	defer client.Close()
+	storageSubID := subID + "-cloud-storage"
+
+	topic, err := getOrCreateTopic(ctx, client, topicID)
+	if err != nil {
+		t.Fatalf("CreateTopic: %v", err)
+	}
+	var buf bytes.Buffer
+
+	// Use the same bucket across test instances. This
```

Comment:
IMO for storage tests we prefer to create/cleanup buckets in each test run; it's not expensive and having dependency on shared resources can lead to flakiness if someone accidentally deletes something.

Reply:
We used to do this with BigQuery subscriptions (using bq datasets/tables), but sometimes when the test invocations crash, cleanup doesn't happen properly and it leads to orphaned resources. This would require us to have a cleanup method that runs prior to sample invocation to delete these resources. I was proposing this change since our samples doesn't really depend on the resource other than it existing, and we guarantee it exists by calling `createOrGet` prior to using it.

Reply:
There is also `testutil.DeleteExpiredBuckets` which can be used to handle this automatically, but ultimately up to you